1.24.0 can not be crosscompiled due to libbpf (externaldep)

##### Bug report summary
```
checking if perf.plugin should be enabled... yes
checking for libelf... yes
checking for struct bpf_prog_info... yes
checking for externaldeps/libbpf/libbpf.a... configure: error: cannot check for file existence when cross compiling
```

##### OS / Environment
My OS:
```
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04.1 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04.1 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04.1 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal
```

##### Steps To Reproduce

1. Download SDK for OpenWrt
2. Use this Makefile https://github.com/openwrt/packages/commit/2de28dd95b8c92e2e76038314cab2e180a4cd554
and change version and hash

##### Expected behavior
It should be compiled even when cross compiling, because package libbpf is not available in OpenWrt
https://github.com/openwrt/packages/pull/12949 

Also, why ebpf plugin can not be disabled as others like nfacct/freeipmi and so on?

--- a/configure.ac
+++ b/configure.ac
@@ -167,6 +167,12 @@ AC_ARG_ENABLE(
     ,
     [enable_jsonc="detect"]
 )
+AC_ARG_ENABLE(
+    [ebpf],
+    [AS_HELP_STRING([--disable-ebpf], [Disable eBPF support @<:@default autodetect@:>@])],
+    ,
+    [enable_ebpf="detect"]
+)
 
 # -----------------------------------------------------------------------------
 # Check if cloud is enabled and if the functionality is available
@@ -968,6 +974,7 @@ AC_CHECK_FILE(
 
 AC_MSG_CHECKING([if ebpf.plugin should be enabled])
 if test "${build_target}" = "linux" -a \
+        "${enable_ebpf}" != "no" -a \
         "${have_libelf}" = "yes" -a \
         "${have_bpf}" = "yes" -a \
         "${have_libbpf}" = "yes"; then

--- a/netdata-installer.sh
+++ b/netdata-installer.sh
@@ -299,7 +299,7 @@ while [ -n "${1}" ]; do
     "--disable-telemetry") NETDATA_DISABLE_TELEMETRY=1 ;;
     "--disable-go") NETDATA_DISABLE_GO=1 ;;
     "--enable-ebpf") NETDATA_DISABLE_EBPF=0 ;;
-    "--disable-ebpf") NETDATA_DISABLE_EBPF=1 ;;
+    "--disable-ebpf") NETDATA_DISABLE_EBPF=1 NETDATA_CONFIGURE_OPTIONS="${NETDATA_CONFIGURE_OPTIONS//--disable-ebpf/} --disable-ebpf" ;;
     "--disable-cloud")
       if [ -n "${NETDATA_REQUIRE_CLOUD}" ]; then
         echo "Cloud explicitly enabled, ignoring --disable-cloud."
@@ -1432,6 +1432,12 @@ should_install_ebpf() {
   if [ "${NETDATA_DISABLE_EBPF:=0}" -eq 1 ]; then
     run_failed "eBPF explicitly disabled."
     defer_error "eBPF explicitly disabled."
+    return 1
+  fi
+
+  if [ "$(uname -s)" != Linux ]; then
+    run_failed "Currently eBPF is only supported on Linux."
+    defer_error "Currently eBPF is only supported on Linux."
     return 1
   fi
 

--- a/netdata-installer.sh
+++ b/netdata-installer.sh
@@ -1435,7 +1435,7 @@ should_install_ebpf() {
     return 1
   fi
 
-  if [ "$(uname -s)" != Linux ]; then
+  if [ "$(uname -s)" != "Linux" ]; then
     run_failed "Currently eBPF is only supported on Linux."
     defer_error "Currently eBPF is only supported on Linux."
     return 1

